### PR TITLE
Form helper

### PR DESCRIPTION
**Pull request by jlangevin**
Replacement for PR #660 - abstract form component in formHelper.js.

### CODE REVIEW

File: src/forms/Predictions.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { Form, Divider, Header, Tab, Message, Button, Menu } from 'semantic-ui-react';
+import { Divider, Header, Tab, Message, Button, Menu } from 'semantic-ui-react';
```
[the remark this answers is not in the document]
Comment (jlangevin) [reply src/forms/Predictions.js]: Good question.  It's being used for things like <Form.Field />

File: src/forms/Predictions.js [hunk 2]
```diff
@@ -134,47 +134,46 @@ const PredictionsStep = function ({ updateClientValue, navData, client, snippets
...
-        <TabbedVisualizations 
-          client   = { client }
-          snippets = { snippets } />
-      </FormPartsContainer>
-    </Form>
+    <FormPartsContainer
+      title     = { snippets.i_title }
+      clarifier = { null }
+      navData   = { navData }
+      formClass = { `income-form` }>
```
Comment (knod) [inline src/forms/Predictions.js]: I think this also needs updating.

File: src/forms/CurrentBenefits.js [hunk 3]
```diff
@@ -69,20 +68,17 @@ const CurrentBenefitsContent = ({ current, updateClientValue, snippets }) => {
...
-          current      = { client.current }
-          snippets     = { snippets } />
-      </FormPartsContainer>
-
-    </Form>
+    <FormPartsContainer
+      title     = { snippets.i_currentBenefits }
+      clarifier = { snippets.i_selectBenefits }
+      navData   = { navData }
+      formClass = { `household-size-form` }
```
[the remark this answers is not in the document]
Comment (jlangevin) [reply src/forms/CurrentBenefits.js]: I simply reused what was already there.  What class should it be?  current-benefits-form would be an obvious guess.  Easy enough to swap it out obviously.   But perhaps a more important question is what's the purpose of these classes to begin with?  A search of the code base does not yield any instance of these form classes being assigned styles.  In theory they could be removed entirely until actually needed at a future date.

[the remark this answers is not in the document]
Comment (knod) [reply src/forms/CurrentBenefits.js]: It's true, we could get rid of them for now. At the time when I implemented a rough version early on, I was talking to some folks about making different colors for different sections or questions. It was one of the ways I was thinking of fighting that 'earned from childcare' question. We can get rid of them for now if we want.

[the remark this answers is not in the document]
Comment (jlangevin) [reply src/forms/CurrentBenefits.js]: In the interest of addressing the original PR that this one replaced, let's leave them in there.  Removing can be a separate PR if we decide we'll never go down the different colors route.

File: src/test/forms/__snapshots__/CurrentBenefits.test.js.snap [hunk 4]
```diff
@@ -1,142 +1,138 @@
...
+<FormPartsContainer
+  clarifier={
+    <span
+      className="snippet"
+      lang="en"
+    >
+      Select the benefits you currently receive
+    </span>
+  }
+  formClass="household-size-form"
```
[the remark this answers is not in the document]
Comment (jlangevin) [reply src/test/forms/__snapshots__/CurrentBenefits.test.js.snap]: This has been updated in a later push.

File: src/forms/Predictions.js (hunk 2 above)
Comment (jlangevin) [reply src/forms/Predictions.js]: This has been updated in a later push.

File: src/forms/CurrentBenefits.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (knod) [reply src/forms/CurrentBenefits.js]: Sure!

**Pull request merged**